File: typescript/raindrops/raindrops.ts
class Raindrops {
  convert(num: number): string {
    let drops = ''
    if (num % 3 === 0) { drops += 'Pling' }
    if (num % 5 === 0) { drops += 'Plang' }
    if (num % 7 === 0) { drops += 'Plong' }
    return drops || String(num)
  }
}

export default Raindrops

File: dart/raindrops/lib/raindrops.dart
class Raindrops {
  String convert(int num) {
    var drops = StringBuffer();
    if (num % 3 == 0) drops.write('Pling');
    if (num % 5 == 0) drops.write('Plang');
    if (num % 7 == 0) drops.write('Plong');
    if (drops.isEmpty) drops.write(num);
    return drops.toString();
  }
}
